### PR TITLE
update(framework): initial version of the python framework

### PR DESCRIPTION
**Pull request by Diogo21Costa**
The framework underwent significant changes, particularly in the system build process, which was completely restructured using Nix for management. In doing so, the code related to the old system build was removed in commit 3c1e043c72a16aa7341e66e43eb089c6335475d4. Additionally, the responsibility for launching the test platform (currently qemu-aarch64-virt) was transferred to the Nix script, resulting in the removal of the previously employed code in commit f7c40ecbb9233c1773c699bbeadb35efd2c17fb9. Furthermore, as part of these modifications, unnecessary processes were eliminated in commit d0b00867840a935a391a6db86f2551a9fcbde401.

## Integration and Logging:

The connection between the test target platform and the Python framework has not been integrated into this version yet. Commit cc2e4f372a4b8a9076bc4df348a8216917c6ae8e removed the script responsible for managing this connection, but it will be re-integrated in a future version. Additionally, the logging monitor was removed in commit 052352dfb8583eea8b69907d8008668a42a4c53a.

## Configuration Changes:

Moving forward, the configuration of the framework will no longer utilize YAML files. Instead, .dts files will be used for configuration in a future version. Consequently, commit 3c1e043c72a16aa7341e66e43eb089c6335475d4 removes the YAML files and the Python scripts that were previously used to parse them.

## New Framework Name:

Finally, I framework's name from the generic "main.py" to a more descriptive name: "test_framework.py" - this change was made in commit d892df5d19e7309a31b024f39cb0a73d3c7f2829.

Please review this PR and provide your feedback.

### CODE REVIEW

File: framework/constants.py
Comment (ESCristiano) [inline framework/constants.py:1]: @Diogo21Costa you should standardize the comments, either use  **# Comment X** or  **""" Comment X """**.

File: framework/test_framework.py
Comment (ESCristiano) [inline framework/test_framework.py:1]: @Diogo21Costa, same comment,  you should standardize the comments, either use **# Comment X** or **""" Comment X """**.

File: framework/constants.py
Comment (Diogo21Costa) [reply framework/constants.py:1]: Thank you for your review! I wanted to clarify the difference between the comment styles used in the code:

1. The triple-quoted (""" Comment X """) comment at the start is a module docstring. It provides an overview of the module's purpose.
2. The single-line comments (# Comment X) explain specific lines of code or steps for developers. In this case, they serve to specify the license.

Both styles serve distinct purposes, enhancing readability. Maintaining this differentiation helps convey both the module's intention and code specifics effectively.

File: framework/test_framework.py
Comment (Diogo21Costa) [reply framework/test_framework.py:1]: I appreciate your feedback! Can you please check the answer to the previous comment? :)

File: framework/constants.py
Comment (ESCristiano) [reply framework/constants.py:1]: Thank you @Diogo21Costa for the clarification. It makes sense! No more concerns from my side. I'll approve.

**Review by ESCristiano — APPROVED** [review]

**Review by miguelafsilva5 — APPROVED** [review]

**Pull request merged**